pennylane/capture/__init__.py

```diff
@@ -0,0 +1,44 @@
+# Copyright 2024 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""
+.. currentmodule:: pennylane
+
+This module implements PennyLane's capturing mechanism for hybrid
+quantum-classical programs.
+
+.. warning::
+
+    This module is experimental and will change significantly in the future.
+
+
+To activate and deactivate the new PennyLane program capturing mechanism, use
+the switches ``qml.capture.enable_plxpr`` and ``qml.capture.disable_plxpr``.
+Whether or not the capturing mechanism is currently being used can be
+queried with ``qml.capture.plxpr_enabled``.
+By default, the mechanism is disabled:
+
+.. code-block:: pycon
+
+    >>> import pennylane as qml
+    >>> qml.capture.plxpr_enabled()
+    False
+    >>> qml.capture.enable_plxpr()
+    >>> qml.capture.plxpr_enabled()
+    True
+    >>> qml.capture.disable_plxpr()
+    >>> qml.capture.plxpr_enabled()
+    False
```

Comment:
What is the intention behind these switches? Do we need them in Q2?

Reply:
I think it's reasonable to make this an opt-in feature right away and to allow all follow-up work to trigger based on that opt-in choice. In addition, JAX is required for plxpr, and it is not a requirement for PennyLane installation currently. The switch provides a concious point in a user program where they would encounter the problem of "I want plxpr but I don't have JAX installed"

Reply:
We are hijacking what happens we create an operation when jitting. 

While PennyLane behaves the same in the vast majority of cases while this is enabled, I did see failures occuring during jax-jit execution.

If we were jitting a normal workflow, and an operator/ measurement intermediary was created, that intermediary would now be an abstract version, rather than the concrete version that we needed.   If we eliminated the use of measurement process intermediaries during jit-execution, we would no longer need these toggles.

So we could get rid of them, but it would be a little more work that I don't think we need to do right now.